lib/parseUri/index.js

```diff
@@ -31,9 +31,13 @@ parseUri.options = {
     name: 'queryKey',
     parser: /(?:^|&)([^&=]*)=?([^&]*)/g
   },
+  // Both of these parser regexes have been edited from the original linked above.
+  // A / character has been added to the excluded characters in the 4th and 5th
+  // capture groups to fix a bug where it would see a @ character in a query
+  // and consume it like it was delimiting a userInfo section.
   parser: {
-    strict: /^(?:([^:\/?#]+):)?(?:\/\/((?:(([^:@]*)(?::([^:@]*))?)?@)?([^:\/?#]*)(?::(\d*))?))?((((?:[^?#\/]*\/)*)([^?#]*))(?:\?([^#]*))?(?:#(.*))?)/,
-    loose: /^(?:(?![^:@]+:[^:@\/]*@)([^:\/?#.]+):)?(?:\/\/)?((?:(([^:@]*)(?::([^:@]*))?)?@)?([^:\/?#]*)(?::(\d*))?)(((\/(?:[^?#](?![^?#\/]*\.[^?#\/.]+(?:[?#]|$)))*\/?)?([^?#\/]*))(?:\?([^#]*))?(?:#(.*))?)/
+    strict: /^(?:([^:\/?#]+):)?(?:\/\/((?:(([^\/:@]*)(?::([^\/:@]*))?)?@)?([^:\/?#]*)(?::(\d*))?))?((((?:[^?#\/]*\/)*)([^?#]*))(?:\?([^#]*))?(?:#(.*))?)/,
```

Comment:
Seems like a reasonable fix to me.  

For consistency with the other character classes, I suggest using ``[^:\/@]`` instead of ``[^\/:@]``.  

Also, should we not include `?` and `#` as excluded characters?  Is ``https://www.example.com?user=a@b.com`` a valid URL??

Reply:
👍 I also confirmed this in IE 11.